tools/clang/lib/Headers/hlsl/dx/linalg.h

```diff
@@ -43,14 +43,24 @@ enum MatrixLayout {
 // Helper for signedness
 //
 namespace details {
-template <typename T> bool IsUnsigned() { return false; }
 
-#ifdef __HLSL_ENABLE_16_BIT
-template <> bool IsUnsigned<uint16_t>() { return true; }
-#endif
+template <typename T> struct IsUnsigned {
+  static const bool Value = false;
```

Comment:
I feel like the default template shouldn't have any definition for `value`, if possible.  That way, unsupported types trigger an error here instead of defaulting to signed.  That would require specializations for float types and all supported signed int types, of course, but it would also prevent use of unexpected types.

Reply:
Do we still want this, given the builtins are now updated to use only the allowed types?